File: client-api/src/main/java/net/yakclient/client/api/annotation/FieldInjection.java
package net.yakclient.client.api.annotation;

import net.yakclient.client.api.annotation.processor.InjectionMetadata;
import net.yakclient.client.api.annotation.processor.InjectionOption;
import net.yakclient.client.api.annotation.processor.InjectionPriorityOption;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
@InjectionMetadata("field")
public @interface FieldInjection {
    @InjectionOption("name")
    String name(); // The field name

    @InjectionOption("access")
    String access(); // The field access

    @InjectionOption("type")
    int type(); // The field type

    @InjectionOption("signature")
    String signature(); // The field signature

    @InjectionPriorityOption
    int priority() default 0;
}

File: client-api/src/main/kotlin/net/yakclient/client/api/annotation/FieldInjection.kt
package net.yakclient.client.api.annotation

import net.yakclient.client.api.annotation.processor.InjectionMetadata
import net.yakclient.client.api.annotation.processor.InjectionOption
import net.yakclient.client.api.annotation.processor.InjectionPriorityOption

// Injects a field into the given class

@InjectionMetadata("field")
public annotation class FieldInjection(
    @InjectionOption("name")
    public val name: String, // The field name
    @InjectionOption("access")
    public val access: String, // The field access
    @InjectionOption("type")
    public val type: Int, // The field type
    @InjectionOption("signature")
    public val signature: String, // The field signature
    @InjectionOption("value")
    public val value: String, // Initial value of the field, must be a primitive type

    @InjectionPriorityOption
    public val priority: Int
)
